### ISSUE TITLE
reconfirmable ignored

### ISSUE THREAD

**Issue by whoteaches** [plataformatec/devise#4022]
## The short story:

I activated config.reconfirmable = true, but when I update a user, it updates the email without copying email to unconfirmed_email and it doesn't email anything. Basically, it does not respect config.reconfirmable = true. I restarted the server, and was trying to debug for two days.
## The details:

I am not able to make devise send reconfirmation emails. It sends sign up confirmation emails just fine, but when I update the user's email, it does not work for me.

I reverted to Devise's own registration controller by removing all my custom code to debug this

```
# registrations_controller.rb
class Users::RegistrationsController < Devise::RegistrationsController
end
```

and even suspended my own user edit view, so it's now using the basic Devise Edit form.

I have this in devise.rb initializer:

```
config.reconfirmable = true
config.allow_unconfirmed_access_for = 2.days and confirmation_sent_at = Time.now

```

My database fields:

```
# user.rb
field :confirmation_token,                 type: String
field :confirmed_at,                       type: Time
field :confirmation_sent_at,               type: Time
field :unconfirmed_email,                  type: String
```

and I have these Devise settings:

```
devise :database_authenticatable, :registerable,
       :recoverable, :rememberable, :trackable,
       :confirmable,
       :omniauthable, :omniauth_providers => [:facebook]

```

My routes are set as follows:

```
devise_for :users, 
  controllers: { 
    omniauth_callbacks: 'users/omniauth_callbacks', 
    registrations: 'users/registrations', 
    sessions: 'users/sessions'
  }
```

The unconfirmed_email never changes, regardless of how often I update the email. The email gets updated actually, which is not logical if it was not confirmed.

What am I missing? How do I debug this?

**Comment by nashby** [comment]
What Devise version are you using? Can you please provide a sample application that reproduces the error?

**Comment by whoteaches** [comment]
I'm using devise 3.5.6

**Comment by ulissesalmeida** [comment]
@whoteaches Are you still having this issue? Could you provide a sample application only with rails and devise that reproduces the error?

**Comment by ulissesalmeida** [comment]
@whoteaches I'm closing this issue. Feel free to reopen when you can evidence the error with sample rails app.

Best

**Comment by Unknown-Guy** [comment]
I'm having the same problem on devise (3.5.8) and also after upgrading to devise (4.1.0)

**Comment by Unknown-Guy** [comment]
Rolled back to 3.5.2 which we had previously and no problem there.

**Comment by ulissesalmeida** [comment]
Thanks, good to know. I'm reopening this issue. If anyone knows how to reproduce feel free to share with us.

**Comment by shaneog** [comment]
Having this same issue after upgrading to 4.1.0 from 4.0.1.

I'll try to put a small PoC Rails app together this week to isolate the issue.

**Comment by ulissesalmeida** [comment]
@shaneog That's weird the release `4.1.0` the only thing that changed about the `reconfirmable` config is its default value (now is true). Did you have news about this issue?

**Comment by shaneog** [comment]
Hi @ulissesalmeida - the error is definitely occurring for me, and only with 4.1.x (and not 4.0.3), but it's to do with something _unique_ in my controllers I suspect.

**Comment by shaneog** [comment]
@ulissesalmeida After spending most of today digging into this issue (it only happens in my test (rspec) environment, in development and production it's fine 😩 ), it turns out that someone better than me has fixed it in #4125. 👍
